pandas/indexes/base.py

```diff
@@ -86,6 +86,12 @@ def _new_Index(cls, d):
     """ This is called upon unpickling, rather than the default which doesn't
     have arguments and breaks __new__
     """
+    # required for backward compat, because PI can't be instantiated with
+    # ordinals through __new__ GH #13277
+    if issubclass(cls, ABCPeriodIndex):
+        if d['data'].dtype == 'int64':
+            values = d.pop('data')
```

Comment:
I think this just a historical reason, originall ``Index`` was a sub-class of ``ndarray``, which used ``__reduce__``. IIRC I changed ``Series`` a while back (these use ``__getstate__/__setstate__``, defined in ``pandas.core.generic``, with some helpers for compat.

So short answer is yes, we should probably change this. But for *this* PR, let's not, pls open a new issue though for this pickle change.